### DISCUSSION TITLE
How to set value of controlled element

### DISCUSSION THREAD

**Discussion by mustanish** [react-hook-form/react-hook-form#3406] [Q&A] [answered]
I am using material-UI with RHF and I have a scenario where my form comes prefilled on page load if data exist in the Redux store.

My component looks something like this

```
`const FormInput:React.FC<InputType> = (props) => {
  const { name, label, type, required, errmsg, control, onChange, value } = props;
  return (
    <Controller 
      name={name} 
      render={
        (props) => <TextField
        onChange={(event) => {
          props.onChange(event);
          onChange(event);
        }}
        value={props.value || value}
        error={errmsg ? true : false} 
        helperText={errmsg} 
        name={name}
        label={label}
        type={type}
        fullWidth={true}
        required={required}
        />
      }
      control={control}
    />
  )
}`

```
I am passing the value as props and inside render method setting value with OR condition. I am able to render the form with the value from the Redux store but if do not edit the form and try to submit, it throws errors that the form is empty. How do I set props.value with value passed as props.

**Comment by bluebill1049** [comment, marked as answer]
if you want to update single input use `setValue`, or entire form with `reset({ ... })`

**Reply by mustanish** [reply]
I want to rerender the entire form with an updated Redux store value. I will try using `reset({ ... })`
